lib/spack/llnl/util/symlink.py

```diff
@@ -66,27 +68,27 @@ def _win32_junction(path, link):
 
 @lang.memoized
 def _win32_can_symlink():
-    tempdir = tempfile.mkdtemp()
+    tempdir = Path(tempfile.mkdtemp())
 
-    dpath = join(tempdir, "dpath")
-    fpath = join(tempdir, "fpath.txt")
+    dpath = tempdir / "dpath"
+    fpath = tempdir / "fpath.txt"
 
-    dlink = join(tempdir, "dlink")
-    flink = join(tempdir, "flink.txt")
+    dlink = tempdir / "dlink"
+    flink = tempdir / "flink.txt"
 
     import llnl.util.filesystem as fs
 
     fs.touchp(fpath)
```

Comment:
This method looks broken: shouldn't we also be creating `dpath`? Also, the parent directory should already exist, so we shouldn't need to call `touchp()` or `mkdirp()`: 
```suggestion
    fpath.touch()
    dpath.mkdir()
```

Reply:
Symlinks are getting a fairly massive overhaul in #34701 so we may want to hold off on function details for this module until that lands.

Reply:
That seems quite reasonable!